### PR TITLE
New: [NA] [Style] [Accordion] Created a Accordion section [ED-10521]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**

### CODE REVIEW

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 1]
```diff
@@ -28,12 +35,13 @@
 				flex-direction: row;
 				list-style: none;
 				font-size: var(--n-accordion-title-font-size);
-				padding: var(--n-accordion-title-padding);
-				border-width: var(--n-accordion-border-width);
+			  	padding: var(--n-accordion-padding) ;
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:38]:
```suggestion
			  	padding: var(--n-accordion-padding);
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 2]
```diff
@@ -57,6 +65,10 @@
 			}
 
 			&[open] {
+			  	.e-n-accordion-item-title {
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:68]:
```suggestion

			  	.e-n-accordion-item-title {
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 3]
```diff
@@ -249,6 +251,149 @@ protected function register_controls() {
 			]
 		);
 		$this->end_controls_section();
+		$this->add_style_tab();
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:255]:
```suggestion
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 4]
```diff
@@ -249,6 +251,148 @@ protected function register_controls() {
...
 		);
 		$this->end_controls_section();
+		$this->add_style_tab();
+	}
+
+	private function add_style_tab() {
+
+		$this->add_accordion_style_section();
+
+	}
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:261]:
```suggestion
	private function add_style_tab() {
		$this->add_accordion_style_section();
	}
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 5]
```diff
@@ -249,6 +251,145 @@ protected function register_controls() {
...
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
+			],
+		]);
+
+		$this->start_controls_tabs( 'accordion__border_and_background' );
+
+		foreach ( array( 'normal', 'hover', 'active' ) as &$state ) {
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:309]:
```suggestion
		foreach ( [ 'normal', 'hover', 'active' ] as &$state ) {
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 6]
```diff
@@ -249,6 +251,145 @@ protected function register_controls() {
...
+
+		$this->end_controls_section();
+	}
+
+
+	/**
+	 * @string $state
+	 */
+	private function add_border_and_radius_style( $state ) {
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:348]:
```suggestion
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 7]
```diff
@@ -249,6 +251,144 @@ protected function register_controls() {
...
+		$this->add_group_control(
+			Group_Control_Border::get_type(),
+			[
+				'name' => 'accordion_border_' . $state,
+				'selector' => $selector,
+			]
+		);
+
+		$this->end_controls_tab();
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:391]:
```suggestion
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 8]
```diff
@@ -249,6 +251,143 @@ protected function register_controls() {
...
+				'size_units' => [ 'px', '%', 'em', 'rem', 'custom' ],
+				'selectors' => [
+					'{{WRAPPER}} ' => '--n-accordion-padding: {{TOP}}{{UNIT}} {{RIGHT}}{{UNIT}} {{BOTTOM}}{{UNIT}} {{LEFT}}{{UNIT}};',
+				],
+			]
+		);
+
+		$this->end_controls_section();
+	}
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:342]:
```suggestion
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 9]
```diff
@@ -17,6 +20,10 @@
 			flex-direction: column;
 			position: relative;
 
+			&:not(:last-child) {
+			  margin-bottom: var(--n-accordion-item-title-space-between);
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:24]:
```suggestion
			    margin-bottom: var(--n-accordion-item-title-space-between);
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 10]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+				$selector .= ':hover';
+				$translated_tab_text = esc_html__( 'Hover', 'elementor' );
+				break;
+			case 'active':
+				$selector = '{{WRAPPER}} .e-n-accordion-item[open] .e-n-accordion-item-title';
+				$translated_tab_text = esc_html__( 'Active', 'elementor' );
+				break;
+		}
+
+		$this->start_controls_tab('accordion_' . $state . '_border_and_background', [
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:360]:
```suggestion
		$this->start_controls_tab( 'accordion_' . $state . '_border_and_background', [
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 11]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+				'name' => 'accordion_background_' . $state,
+				'types' => [ 'classic', 'gradient' ],
+				'exclude' => [ 'image' ],
+				'fields_options' => [
+					'color' => [
+						'label' => esc_html__( 'Color', 'elementor' ),
+					],
+				],
+				'selector' => $selector,
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:376]:
```suggestion
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 12]
```diff
@@ -249,6 +251,140 @@ protected function register_controls() {
...
+			],
+			'default' => [
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
+			],
+		]);
+
+		$this->start_controls_tabs( 'accordion__border_and_background' );
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:306]:
```suggestion
		$this->start_controls_tabs( 'accordion_border_and_background' );
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 13]
```diff
@@ -17,6 +20,10 @@
 			flex-direction: column;
 			position: relative;
 
+			&:not(:last-child) {
+			    margin-bottom: var(--n-accordion-item-title-space-between);
+			}
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:25]:
```suggestion
			&:not(:last-child) {
					margin-bottom: var(--n-accordion-item-title-space-between);
			}
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 14]
```diff
@@ -17,6 +20,10 @@
 			flex-direction: column;
 			position: relative;
 
+			&:not(:last-child) {
+					margin-bottom: var(--n-accordion-item-title-space-between);
+			}
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:25]:
```suggestion
			&:not(:last-child) {
				margin-bottom: var(--n-accordion-item-title-space-between);
			}
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 15]
```diff
@@ -249,6 +251,140 @@ protected function register_controls() {
 			]
 		);
 		$this->end_controls_section();
+		$this->add_style_tab();
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:254]:
```suggestion

		$this->add_style_tab();
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 16]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+			],
+			'default' => [
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-space-between: {{SIZE}}{{UNIT}}',
+			],
+		]);
+
+		$this->add_responsive_control('accordion_item_title_distance_from_content', [
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:289]:
```suggestion
		$this->add_responsive_control( 'accordion_item_title_distance_from_content', [
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 17]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
+			],
+		]);
+
+		$this->start_controls_tabs( 'accordion_border_and_background' );
+
+		foreach ( [ 'normal', 'hover', 'active' ] as &$state ) {
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:309]: @TzviRabinovitch Why is `&$state` being passed by reference?

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 18]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+	private function add_accordion_style_section() {
+		$this->start_controls_section(
+			'section_accordion_style',
+			[
+				'label' => esc_html__( 'Accordion', 'elementor' ),
+				'tab' => Controls_Manager::TAB_STYLE,
+			]
+		);
+
+		$this->add_responsive_control('accordion_item_title_space_between', [
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:271]:
```suggestion
		$this->add_responsive_control( 'accordion_item_title_space_between', [
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 19]
```diff
@@ -249,6 +251,141 @@ protected function register_controls() {
...
+					'max' => 200,
+				],
+			],
+			'default' => [
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-space-between: {{SIZE}}{{UNIT}}',
+			],
+		]);
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:287]:
```suggestion
		$this->add_responsive_control(
		        'accordion_item_title_space_between', [
			'label' => esc_html__( 'Space between Items', 'elementor' ),
			'type' => Controls_Manager::SLIDER,
			'size_units' => [ 'px' ],
			'range' => [
				'px' => [
					'min' => 0,
					'max' => 200,
				],
			],
			'default' => [
				'size' => 0,
			],
			'selectors' => [
				'{{WRAPPER}}' => '--n-accordion-item-title-space-between: {{SIZE}}{{UNIT}}',
			],
		]);
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 20]
```diff
@@ -249,6 +251,142 @@ protected function register_controls() {
...
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:288]:
```suggestion
		$this->add_responsive_control(
			'accordion_item_title_space_between',
			[
				'label' => esc_html__( 'Space between Items', 'elementor' ),
				'type' => Controls_Manager::SLIDER,
				'size_units' => [ 'px' ],
				'range' => [
					'px' => [
						'min' => 0,
						'max' => 200,
					],
				],
				'default' => [
					'size' => 0,
				],
				'selectors' => [
					'{{WRAPPER}}' => '--n-accordion-item-title-space-between: {{SIZE}}{{UNIT}}',
				],
			]
		);
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 21]
```diff
@@ -249,6 +251,144 @@ protected function register_controls() {
...
+					'max' => 200,
+				],
+			],
+			'default' => [
+				'size' => 0,
+			],
+			'selectors' => [
+				'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
+			],
+		]);
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:308]:
```suggestion
		$this->add_responsive_control(
			'accordion_item_title_distance_from_content',
			[
				'label' => esc_html__( 'Distance from content', 'elementor' ),
				'type' => Controls_Manager::SLIDER,
				'size_units' => [ 'px' ],
				'range' => [
					'px' => [
						'min' => 0,
						'max' => 200,
					],
				],
				'default' => [
					'size' => 0,
				],
				'selectors' => [
					'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
				],
			]
		);
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 22]
```diff
@@ -249,6 +251,147 @@ protected function register_controls() {
...
+				break;
+			case 'active':
+				$selector = '{{WRAPPER}} .e-n-accordion-item[open] .e-n-accordion-item-title';
+				$translated_tab_text = esc_html__( 'Active', 'elementor' );
+				break;
+		}
+
+		$this->start_controls_tab( 'accordion_' . $state . '_border_and_background', [
+			'label' => $translated_tab_text,
+		]);
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:369]:
```suggestion
		$this->start_controls_tab(
			'accordion_' . $state . '_border_and_background',
			[
				'label' => $translated_tab_text,
			]
		);
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 23]
```diff
@@ -249,6 +251,150 @@ protected function register_controls() {
...
+				],
+				'selectors' => [
+					'{{WRAPPER}}' => '--n-accordion-item-title-distance-from-content: {{SIZE}}{{UNIT}}',
+				],
+			]
+		);
+
+		$this->start_controls_tabs( 'accordion_border_and_background' );
+
+		foreach ( [ 'normal', 'hover', 'active' ] as &$state ) {
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:315]:
```suggestion
		foreach ( [ 'normal', 'hover', 'active' ] as $state ) {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 24]
```diff
@@ -9,6 +8,10 @@
 	--n-accordion-item-title-justify-content-mobile: initial;
 	--n-accordion-title-icon-order: -1;
 	--n-accordion-title-icon-padding-inline: 0 15px;
+  	--n-accordion-item-title-space-between: 0px;
+  	--n-accordion-item-title-distance-from-content: 0px;
```
Comment (Barcov) [inline modules/nested-accordion/assets/scss/frontend.scss:12]: @TzviRabinovitch can we shorten these class names to:
`--n-accordion-title-space-between:`
`--n-accordion-title-distance-from-content`

**Review by ArmanPurtoian — APPROVED** [review]

**Review by louiswol94 — APPROVED** [review]

**Review by Barcov — APPROVED** [review]

**Pull request merged**